api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2AuthenticatorConfig.java

```diff
@@ -143,11 +146,26 @@ static OAuth2AuthenticatorConfig fromConfigSupplier(Function<String, String> con
         CONF_NESSIE_OAUTH2_DEVICE_CODE_FLOW_POLL_INTERVAL,
         builder::deviceCodeFlowPollInterval,
         Duration::parse);
+    applyConfigOption(
+        config,
+        CONF_NESSIE_OAUTH2_EXTRA_PARAMS,
+        builder::extraRequestParameters,
+        OAuth2AuthenticatorConfig::parseExtraParams);
     builder.tokenExchangeConfig(TokenExchangeConfig.fromConfigSupplier(config));
     builder.impersonationConfig(ImpersonationConfig.fromConfigSupplier(config));
     return builder.build();
   }
 
+  static Map<String, String> parseExtraParams(String text) {
+    if (text == null || text.isBlank()) {
+      return Map.of();
+    }
+    return Arrays.stream(text.split(","))
+        .map(s -> s.split("=", 2))
+        .collect(
+            Collectors.toMap(a -> a[0].trim(), a -> a.length > 1 ? a[1].trim() : "", (a, b) -> a));
```

Comment:
Would it make sense to throw on duplicate keys (default merge function)? The order or application of the merge function is not well-defined anyway, so it will be hard to predict the resultant config map values in case of key collisions... WDYT?

Reply:
Good idea but I had to revert to a classical for loop because the `BinaryOperator` that merges duplicate entries does not have access to the offending key, only to the offending values.